Code/Framework/AzCore/AzCore/Task/TaskGraphSystemComponent.cpp

```diff
@@ -28,12 +30,23 @@ namespace AZ
     {
         AZ_Assert(m_taskExecutor == nullptr, "Error multiple activation of the TaskGraphSystemComponent");
 
+        AZ::ApplicationTypeQuery appType;
+        AZ::ComponentApplicationBus::Broadcast(&AZ::ComponentApplicationBus::Events::QueryApplicationType, appType);
+        if (!appType.IsValid() || (appType.IsTool() && !appType.IsEditor()))
```

Comment:
I guess this will affect the Material Editor & the Animation Editor ?

Reply:
Maybe as next step, after this issue unblocks the team, is to provide a IsBuilder() function.

Reply:
Yea, none of those tools are using this functionality at the moment. This system as designed was really only meant to be for the primary simulation/renderer runtime.

Reply:
The Animation Editor is not a standalone process, it runs embedded in the Editor

Reply:
How about just check if the Settings Registry has a specialization key for [assetbuilder](https://github.com/o3de/o3de/blob/development/Code/Tools/AssetProcessor/AssetBuilder/AssetBuilderApplication.cpp#L97-L98)

```
if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
{
    using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
    using Type = AZ::SettingsRegistryInterface::Type;

    auto assetBuilderSpecializationKey = FixedValueString::format("%s/assetbuilder", AZ::SettingsRegistryMergeUtils::SpecializationsRootKey);
    // Check if the assetbuilder specialization exist in the settings registry to validate if the Asset Builder is running
    if (settingsRegistry->GetType(assetBuilderSpecializationKey) != Type::NoType)
    {

    }
}
```

Reply:
It's useful to know this is an option, but for now I think it's ok to restrict concurrency to other exes also.

Reply:
Alternatively you can just make this configurable through a "taskgraph.assetbuilder.setreg" file you can toss in the [Registry/](https://github.com/o3de/o3de/tree/development/Registry) directory.

```JSON
{
    "O3DE": {
        "TaskGraph": {
            "Settings": {
                "GraphThreadsMaxCount": 1
            }
        }
    }
}
```

You can then just add code to query that value from the Settings Registry